docs/cluster-large.md

```diff
@@ -0,0 +1,107 @@
+
+For example:
+```YAML
+      containers:
+        - image: gcr.io/google_containers/heapster:v0.15.0
+          name: heapster
+          resources:
+            limits:
+              cpu: 100m
+              memory: 200Mi
+```
+
+These limits, however, are based on data collected from addons running on 4-node clusters (see [#10335](https://github.com/GoogleCloudPlatform/kubernetes/issues/10335#issuecomment-117861225)). The addons consume a lot more resources when running on large deployment clusters (see [#5880](https://github.com/GoogleCloudPlatform/kubernetes/issues/5880#issuecomment-113984085)). So, if a large cluster is deployed without adjusting these values, the addons may continuously get killed because they keep hitting the limits.
+
+To avoid running into cluster addon resource issues, when creating a cluster with many nodes, consider the following:
+* Scale memory and CPU limits for each of the following addons, if used, along with the size of cluster (there is one replica of each handling the entire cluster so memory and CPU usage tends to grow proportionally with size/load on cluster):
+  * Heapster ([GCM/GCL backed](cluster/addons/cluster-monitoring/google/heapster-controller.yaml), [InfluxDB backed](cluster/addons/cluster-monitoring/influxdb/heapster-controller.yaml), [GCM/InfluxDB](cluster/addons/cluster-monitoring/googleinfluxdb/heapster-controller-combined.yaml), [standalone](cluster/addons/cluster-monitoring/standalone/heapster-controller.yaml))
+  * [InfluxDB and Grafana](cluster/addons/cluster-monitoring/influxdb/influxdb-grafana-controller.yaml)
+  * [skydns, kube2sky, and dns etcd](cluster/addons/dns/skydns-rc.yaml.in)
+  * [Kibana](cluster/addons/fluentd-elasticsearch/kibana-controller.yaml)
+* Scale number of replicas for the following addons, if used, along with the size of cluster (there are multiple replicas of each so increasing replicas should help handle increased load, but, since load per replica also increases slightly, also consider increasing CPU/memory limits):
+  * [elasticsearch](cluster/addons/fluentd-elasticsearch/es-controller.yaml)
+* Increase memory and CPU limits sligthly for each of the following addons, if used, along with the size of cluster (there is one replica per node but CPU/memory usage increases slightly along with cluster load/size as well):
+  * [FluentD with ElasticSearch Plugin](cluster/saltbase/salt/fluentd-es/fluentd-es.yaml)
+  * [FluentD with GCP Plugin](cluster/saltbase/salt/fluentd-gcp/fluentd-gcp.yaml)
+
+
+#### Detecting Resource Starved Containers
```

Comment:
Please move this section to its own document, and link from here, and also link from docs/compute_resources.md.